### PR TITLE
add guild member update username property

### PR DESCRIPTION
**Pull request by zaanposni**
Adds the guild member username before and after properties to the update event args.

### CODE REVIEW

**Comment by Neuheit** [comment]
The user's username is not updated on this event.

**Comment by zaanposni** [comment]
Quoting from the [docs](https://discord.com/developers/docs/topics/gateway#guild-member-update)
> Sent when a guild member is updated. This will also fire when the user object of a guild member changes.

Ohh and it also works when testing:
```cs
_client.GuildMemberUpdated += this.GuildMemberUpdatedHandler;
private Task GuildMemberUpdatedHandler(DiscordClient sender, GuildMemberUpdateEventArgs e)
{
    _logger.LogWarning("this actually triggers");
}
 ```

**Comment by DWaffles** [comment]
I have quickly tested this out and GuildMemberUpdate does fire when a username is changed.

File: DSharpPlus/Clients/DiscordClient.Dispatch.cs [hunk 1]
```diff
@@ -1110,7 +1110,9 @@ internal async Task OnGuildMemberUpdateEventAsync(TransportMember member, Discor
             var pending_old = mbr.IsPending;
             var roles_old = new ReadOnlyCollection<DiscordRole>(new List<DiscordRole>(mbr.Roles));
             var avatar_old = mbr.GuildAvatarHash;
+            var username_old = mbr.Username;
 
+            mbr.Username = member.User.Username;
```
Comment (VelvetThePanda) [inline DSharpPlus/Clients/DiscordClient.Dispatch.cs:1115]: `DiscordMember#Username` is just a passthrough to the user object if I’m not mistaken? Using that would be more consistent.

**Review by VelvetThePanda — COMMENTED** [review]
Small nitpick but looks good otherwise.

**Comment by VelvetThePanda** [comment]
Agh, a transport member. That explains it.

**Comment by VelvetThePanda** [comment]
Currently looking through stale PRs; there's a merge conflict, but I think otherwise this is good.

**Pull request merged**